### PR TITLE
Suggestion on re-arranging the docs

### PR DESCRIPTION
**Pull request by namelessvoid**
This PR is a suggestion for #259 .

Updates to the structure: 

- The "Getting Started" section contains installation instructions for GDExtension version, module version and on C#.
- The "Behavior Trees" and "Hierarchical State Machines" sections contain their respective docs.
- The "Important classes" page was moved to the top of the "Class Reference" section.

I did not remove any info (just moved it - if something is missing, this was unintentional). I added some lines on HSMs on the index page and also added some "filler" text.

The proposed structure looks like this:

image

### CODE REVIEW

**Comment by namelessvoid** [comment]
Note: I did not update the Readme file yet. If you like my proposal, I can update the Readme, too :slightly_smiling_face:

File: doc/source/getting-started/getting-limboai.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+If you're unsure, start with the GDExtension version.
+You can change your decision at any time - both versions are fully compatible.
+
+Get GDExtension version
+------------------------
+
+GDExtension is the most convenient way of using the LimboAI plugin, but it comes
+with certain limitations:
+
+* Built-in documentation is not available. The plugin will open online documentation instead when requested.
```
Comment (limbonaut) [inline doc/source/getting-started/getting-limboai.rst:21]: This is no longer true since 1.2

**Comment by limbonaut** [comment]
There's also "Contributing" page missing that was added recently. You may want to rebase your branch to see it.

**Comment by namelessvoid** [comment]
> There's also "Contributing" page missing that was added recently. You may want to rebase your branch to see it.

True! I did rebase yesterday, but forgot to update the screenshot. The Contributing page is located in the "Getting Started" section.

**Comment by limbonaut** [comment]
Looks good! I just have a few suggestions before we merge it.

**Comment by namelessvoid** [comment]
I like your suggestions and applied them! The Links in the Readme are now updated as well.

**Comment by limbonaut** [comment]
Thanks

**Pull request merged**